internal/pkg/service/buffer/sink/tablesink/storage/repository/file_repo.go

```diff
@@ -308,7 +308,9 @@ func (r *FileRepository) rotateAllIn(rb rollback.Builder, now time.Time, parentK
 
 	// Get all active volumes
 	var volumes []volume.Metadata
-	atomicOp.ReadOp(r.all.Volume().ListWriterVolumes().WithAllTo(&volumes))
+	if openNew {
+		atomicOp.ReadOp(r.all.Volume().ListWriterVolumes().WithAllTo(&volumes))
+	}
 
```

Comment:
Volumes are needed only if we are opening a new file, in `RotateAllInOperation`, but they are not needed for `CloseAllIn` operation.